src/dialect/snowflake.rs

```diff
@@ -137,6 +147,212 @@ pub fn parse_create_stage(
+            from_stage = parser.parse_object_name()?;
+
+            // as
+            from_stage_alias = if parser.parse_keyword(Keyword::AS) {
+                Some(match parser.next_token().token {
+                    Token::Word(w) => Ok(Ident::new(w.value)),
+                    _ => parser.expected("stage alias", parser.peek_token()),
+                }?)
+            } else {
+                None
+            };
+            parser.expect_token(&Token::RParen)?;
+        }
+        _ => {
+            parser.prev_token();
+            from_stage = parser.parse_object_name()?;
+
+            // as
+            from_stage_alias = if parser.parse_keyword(Keyword::AS) {
+                Some(match parser.next_token().token {
+                    Token::Word(w) => Ok(Ident::new(w.value)),
+                    _ => parser.expected("stage alias", parser.peek_token()),
+                }?)
+            } else {
+                None
+            };
+        }
+    };
+
+    // [ files ]
+    if parser.parse_keyword(Keyword::FILES) {
+        parser.expect_token(&Token::Eq)?;
+        parser.expect_token(&Token::LParen)?;
+        let mut continue_loop = true;
+        while continue_loop {
+            continue_loop = false;
+            let next_token = parser.next_token();
+            match next_token.token {
+                Token::SingleQuotedString(s) => files.push(s),
+                _ => parser.expected("file token", next_token)?,
+            };
+            if parser.next_token().token.eq(&Token::Comma) {
+                continue_loop = true;
+            } else {
+                parser.prev_token(); // not a comma, need to go back
+            }
+        }
+        parser.expect_token(&Token::RParen)?;
+    }
+
+    // [ pattern ]
+    let mut pattern = None;
+    if parser.parse_keyword(Keyword::PATTERN) {
+        parser.expect_token(&Token::Eq)?;
+        let next_token = parser.next_token();
+        pattern = Some(match next_token.token {
+            Token::SingleQuotedString(s) => s,
+            _ => parser.expected("pattern", next_token)?,
+        });
+    }
+
+    // [ file_format]
+    let mut file_format = Vec::new();
+    if parser.parse_keyword(Keyword::FILE_FORMAT) {
+        parser.expect_token(&Token::Eq)?;
+        file_format = parse_parentheses_options(parser)?;
+    }
+
+    // [ copy_options ]
+    let mut copy_options = Vec::new();
+    if parser.parse_keyword(Keyword::COPY_OPTIONS) {
+        parser.expect_token(&Token::Eq)?;
+        copy_options = parse_parentheses_options(parser)?;
+    }
+
+    // [ VALIDATION_MODE ]
+    let mut validation_mode = None;
+    if parser.parse_keyword(Keyword::VALIDATION_MODE) {
+        parser.expect_token(&Token::Eq)?;
+        validation_mode = Some(parser.next_token().token.to_string());
+    }
+
+    Ok(Statement::CopyIntoSnowflake {
+        into,
+        from_stage,
+        from_stage_alias,
+        stage_params: StageParamsObject {
```

Comment:
I didn't see any test coverage for stage params -- could you please add some coverage for that as well?

Reply:
Yes, you're right. I've included `test_copy_into_with_stage_params`.